nomad/structs/structs.go

```diff
@@ -7928,15 +7928,10 @@ func (t *Task) Validate(jobType string, tg *TaskGroup) error {
 
 	// Validation for volumes
 	for idx, vm := range t.VolumeMounts {
-		if !MountPropagationModeIsValid(vm.PropagationMode) {
-			mErr.Errors = append(mErr.Errors, fmt.Errorf("Volume Mount (%d) has an invalid propagation mode: \"%s\"", idx, vm.PropagationMode))
-		}
-
-		// Validate the task does not reference undefined volume mounts
-		if vm.Volume == "" {
-			mErr.Errors = append(mErr.Errors, fmt.Errorf("Volume Mount (%d) references an empty volume", idx))
-		} else if _, ok := tg.Volumes[vm.Volume]; !ok {
+		if _, ok := tg.Volumes[vm.Volume]; !ok {
 			mErr.Errors = append(mErr.Errors, fmt.Errorf("Volume Mount (%d) references undefined volume %s", idx, vm.Volume))
+		} else if err := vm.Validate(); err != nil {
+			mErr.Errors = append(mErr.Errors, fmt.Errorf("Volume Mount (%d) is invalid: \"%w\"", idx, err))
```

Comment:
Do you have an example of how this error looks like? I'm not sure why we do it in some many parts of the code, but appending directly to `mErr.Errors` usually results in a bad error hierarchy.

Reply:
it is actually called in the task group validation test, it should not be a problem